spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -712,17 +712,6 @@ class CometSparkSessionExtensions
     }
 
     override def apply(plan: SparkPlan): SparkPlan = {
-      // DataFusion doesn't have ANSI mode. For now we just disable CometExec if ANSI mode is
-      // enabled.
-      if (isANSIEnabled(conf)) {
-        if (COMET_ANSI_MODE_ENABLED.get()) {
-          logWarning("Using Comet's experimental support for ANSI mode.")
-        } else {
-          logInfo("Comet extension disabled for ANSI mode")
-          return plan
-        }
-      }
```

Comment:
For Spark 3.x, I feel we still need this protection in case users enable ANSI mode as well as native execution.
If I understand correctly we have not set up CI yet with ANSI enabled.

Reply:
This current check will make all plans fall back to Spark if ANSI is enabled, so no native plans will run, unless `COMET_ANSI_MODE_ENABLED` is enabled.

The changes in this PR mean that we still have the same check but only if the plan actually contains any expressions that would be affected by enabling ANSI support and we still fall back to Spark by default unless `COMET_ANSI_MODE_ENABLED` is enabled.

The main risk with this PR is if we don't have the complete list of expressions that support ANSI mode.

Reply:
> If I understand correctly we have not set up CI yet with ANSI enabled.

Well, we have the Spark 4 tests where ANSI is enabled and that is catching issues for sure.

I noticed that we were explicitly disabled ANSI mode in CometTestBase and I have removed that so we use the default ANSI mode for whatever Spark version we are running against. This should mean that all of our unit tests will now run with ANSI enabled when running against Spark 4+. Let's see how many issues this finds 🌑

Reply:
> If I understand correctly we have not set up CI yet with ANSI enabled.

It would be good to add a CI pipeline for Spark 3.4 with ANSI enabled.

Reply:
I am close to finish my PR to enable the spark tests with Spark 4.0 I.e. ANSI enabled, and I am disabling expressions that failed those tests with ANSI enabled.

I would propose to hold this part of change for now. Perhaps after we fix all Spark 4.0 issues, we can backport the learning to Spark 3.4

Reply:
Sure. I have moved this to draft for now.